### PR TITLE
Fix missing check for 0 in craft replacements

### LINKED ISSUE #2877: Player files corrupting with "blank" items
Blank items with no description, no texture, and a very large item count are ending up in players' crafting grids when certain crafting actions are taken.

Reproduceable in singleplayer mode or on a server.  Minetest in both cases is at commit 7b171ea2be0e476d7cdc9300b53ba86a9f694161

Not sure when it broke, only saw it happen to a few clients to far as well as myself in singleplayer mode.  Tried rolling the server back to my previous build (commit 29dda9f356042c403b3b7da1d717d32b45c9b6de as indicated by `git log -g`) and to the one before it (3d43df99f1b706c836a43329d224d4b5b9ba5191).  Happens in both of those, as well.

Excerpt from an affected player file:

```
breath = 11
hp = 20
name = Carlh
pitch = 7.5
position = (-19062,-2155,-8800.68)
version = 1
yaw = 48.93
PlayerArgsEnd
List main 32
Width 0
Item default:apple 4
Item default:axe_steel 1 33488
Item default:torch 79

[...]

List craft 9
Width 3
Item gloopblocks:cement 22
Item  65513
Item  65528
Item  65527
Empty
Empty
Item  65513
Item  65522
Item  65513
EndInventoryList

[...]
```

(Note the crafting grid section at the end here)

Steps to reproduce:
- Start with vanilla minetest_game, no mods, at git commit https://github.com/minetest/minetest_game/commit/0914e595c7ab01a66188fca52561df26f852d7c4 
- Create a test world.
- Install https://github.com/VanessaE/gloopblocks and activate it for that world.
- `/grant singleplayer all`
- `/giveme gravel 2`
- `/giveme bucket:bucket_water`
- Put the stack of 2 gravel into one slot in the craft grid
- Put the water bucket next to the cobble stack.  Wet cement will appear on the output
- Take the wet cement out of the result slot and put it in your inventory.  An empty bucket and one gravel block will remain in the craft grid.
- Close the game and examine `minetest/worlds/foo/players/singleplayer`.  You'll find the empty bucket and one gravel in the crafting grid section, and the rest of the entries therein will be filled with `Item  65535` (two spaces).

This would probably work for any recipe that involves a replacement.  I chose the gloopblocks wet cement recipe here because FreeFull suggested via IRC that it might trigger the bug, and besides the recipe is not likely to be the cause. :stuck_out_tongue_closed_eyes: 

These "blank" items can be transferred to the main player inventory as well, by picking a regular item such as cobble from your inventory, dropping into a crafting slot that's occupied by a blank item (thus initiating the usual swap-stacks action), and then dropping the now-attached blank item into your inventory.  Such items can then be transferred to a chest also, using a similar set of actions.

As a guess, 17ba584fe254eeaee3489cc20e03810a59f3ef9b may be the cause, as everything appears to behave normally if I roll back to 0b76e85a717a05b608d09618bb48434510f0ae27.

### PR DESCRIPTION
**Pull request by TeTpaAka**
Fixes  #2877

### CODE REVIEW

File: src/craftdef.cpp [hunk 1]
```diff
@@ -236,7 +236,7 @@ static void craftDecrementOrReplaceInput(CraftInput &input,
 			}
 		}
 		// No replacement was found, simply decrement count to zero
-		if (!found_replacement)
+		if (!found_replacement and it->count > 0)
```
Comment (Zeno-) [inline src/craftdef.cpp]: Please use &&

Comment (TeTpaAka) [inline src/craftdef.cpp]: Fixed.

File: src/craftdef.cpp [hunk 2]
```diff
@@ -236,7 +236,7 @@ static void craftDecrementOrReplaceInput(CraftInput &input,
 			}
 		}
 		// No replacement was found, simply decrement count to zero
```
Comment (est31) [inline src/craftdef.cpp]: Btw, does it always decrement the count **to zero**? From how I see it, it only decrements **by one**. The comment should perhaps be fixed.

Comment (TeTpaAka) [inline src/craftdef.cpp]: The comment was there before. But you're right. Fixed

**Comment by est31** [comment]
Thanks for the quick fix. Merged: 96989e0a6aa3ab069b5aeeab44a6280d6d51364a

**Pull request closed without merging**